src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
-                _ => other._type switch
-                {
-                    InternalType.Infinity         => -1,
-                    InternalType.NegativeInfinity => 1,
-                    _                             => _daysSinceEra.CompareTo(other._daysSinceEra)
-                }
-            };
+        /// <summary>
+        /// The value of this constant is equivalent to January 1, 1970, in the Gregorian calendar.
+        /// <see cref="UnixEpoch"/> defines the point in time when Unix time is equal to 0.
+        /// </summary>
+        public static NpgsqlDate UnixEpoch => new NpgsqlDate(-10957);
 
-        public int CompareTo(object? o)
-            => o == null
-                ? 1
-                : o is NpgsqlDate npgsqlDate
-                    ? CompareTo(npgsqlDate)
-                    : throw new ArgumentException();
+        /// <summary>Gets the year component of the date represented by this instance.</summary>
+        /// <value>The year component.</value>
+        public int Year => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Year);
 
-        public override int GetHashCode() => _daysSinceEra;
+        /// <summary>Gets the month component of the date represented by this instance.</summary>
+        /// <value>The month component, expressed as a value between 1 and 12.</value>
+        public int Month => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Month);
 
-        #endregion
+        /// <summary>Gets the day component of the date represented by this instance.</summary>
+        /// <value>The day component, expressed as a value between 1 and 31.</value>
+        public int Day => NpgsqlDateTime.GetDatePart(Days, NpgsqlDateTimePart.Day);
 
-        #region Operators
+        /// <summary>Gets the number of days since PostgreSQL epoch.</summary>
+        /// <value>The number of days since PostgreSQL epoch.</value>
+        /// <seealso cref="PostgreSqlEpoch"/>
+        public int Days => _days;
 
-        public static bool operator ==(NpgsqlDate x, NpgsqlDate y) => x.Equals(y);
-        public static bool operator !=(NpgsqlDate x, NpgsqlDate y) => !(x == y);
-        public static bool operator <(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) < 0;
-        public static bool operator >(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) > 0;
-        public static bool operator <=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) <= 0;
-        public static bool operator >=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) >= 0;
+        /// <summary>
+        /// Determines whether the specified value is finite.
+        /// </summary>
+        /// <param name="value">A date.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified value is finite;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool IsFinity(NpgsqlDate value) =>
```

Comment:
Any reason for this to be a static method rather than just an instance property?